### ISSUE TITLE
[ML] Filter platform native code when building platform-specific distributions

### ISSUE THREAD

**Issue by droberts195** [elastic/elasticsearch#39229]
Now that Elasticsearch has platform-specific distributions (starting in version 7.0) it doesn't make sense to ship the ML C++ binaries for all platforms in every distribution.

We should change the code that builds the distributions to filter the ML `platform` sub-directories that are not for the correct platform.

(Note: this needs to be done at the point of building the distributions because it is crucial that a single build machine can build the distributions for all platforms.)

**Comment by droberts195** [comment]
One reason to do this is that `lintian` reports errors when scanning the `.deb` package because of the non-Linux binaries in the distribution:

```
W: elasticsearch: executable-not-elf-or-script usr/share/elasticsearch/modules/x-pack-ml/platform/darwin-x86_64/bin/controller
W: elasticsearch: executable-not-elf-or-script usr/share/elasticsearch/modules/x-pack-ml/platform/windows-x86_64/bin/boost_program_options-vc141-mt-1_65_1.dll
W: elasticsearch: executable-not-elf-or-script usr/share/elasticsearch/modules/x-pack-ml/platform/windows-x86_64/bin/vcruntime140.dll
W: elasticsearch: executable-not-elf-or-script ... use --no-tag-display-limit to see all (or pipe to a file/program)
```

**Comment by jasontedor** [comment]
Closed by #41089
